### ISSUE TITLE
eliminating from recursive HIT with recursive boundary

### ISSUE THREAD

**Issue by ecavallo** [RedPRL/redtt#254]
```
import unit

data wow where
| suc (x : wow) 
| loop (x : wow) @ i [
  | i=0 ⇒ suc x
  | i=1 ⇒ suc x
  ]

let test (x : wow) : unit =
  elim x [
  | suc x' ⇒ ?
  | loop x' i ⇒ ?
  ]
```
raises `Failure("eval_head: de bruijn index out of range")`.

**Comment by ecavallo** [comment]
That's on `parametric-hits`; on `master` it raises `Failure("Bwd.nth")`.

**Comment by ecavallo** [comment]
The error disappears when the first hole is replaced with `triv`.

**Comment by ecavallo** [comment]
Resolved by #258.

**Comment by jonsterling** [comment]
Weird, for some reason this doesn't appear fixed? I thought I had fixed it, but I'm confused.
